retrofit2/src/main/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaRetrofitBuilder.java

```diff
@@ -0,0 +1,208 @@
+import static com.linecorp.armeria.client.http.retrofit2.ArmeriaCallFactory.GROUP_PREFIX;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.util.concurrent.Executor;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+
+import com.linecorp.armeria.client.ClientFactory;
+import com.linecorp.armeria.client.ClientOptions;
+import com.linecorp.armeria.client.ClientOptionsBuilder;
+import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.client.endpoint.EndpointGroup;
+import com.linecorp.armeria.client.http.HttpClient;
+import com.linecorp.armeria.common.Scheme;
+import com.linecorp.armeria.common.SerializationFormat;
+import com.linecorp.armeria.common.SessionProtocol;
+
+import okhttp3.HttpUrl;
+import retrofit2.Call;
+import retrofit2.CallAdapter;
+import retrofit2.Callback;
+import retrofit2.Converter;
+import retrofit2.Retrofit;
+import retrofit2.Retrofit.Builder;
+
+/**
+ * A helper class for creating a new {@link Retrofit} instance with {@link ArmeriaCallFactory}.
+ * For example,
+ *
+ * <pre>{@code
+ * Retrofit retrofit = new ArmeriaRetrofitBuilder()
+ *     .baseUrl("http://localhost:8080/")
+ *     .build();
+ *
+ * MyApi api = retrofit.create(MyApi.class);
+ * Response&lt;User&gt; user = api.getUser().execute();
+ * }</pre>
+ *
+ * <p>{@link ArmeriaRetrofitBuilder} even supports {@link EndpointGroup}, so you can create {@link Retrofit}
+ * like below,
+ *
+ * <pre>{@code
+ * EndpointGroupRegistry.register("foo",
+ *                                new StaticEndpointGroup(Endpoint.of("127.0.0.1", 8080)),
+ *                                ROUND_ROBIN);
+ *
+ * Retrofit retrofit = new ArmeriaRetrofitBuilder()
+ *     .baseUrl("http://group:foo/")
+ *     .build();
+ * } </pre>
+ */
+public final class ArmeriaRetrofitBuilder {
+
+    private static final Pattern GROUP_PREFIX_PATTERN = Pattern.compile("^[_0-9a-z]+$");
+    private static final BiFunction<String, ? super ClientOptionsBuilder, ClientOptionsBuilder>
+            DEFAULT_CONFIGURATOR = (url, optionsBuilder) -> optionsBuilder;
+    private static final String SLASH = "/";
+
+    private final Retrofit.Builder retrofitBuilder;
+    private final ClientFactory clientFactory;
+    private String baseUrl;
+    private String basePath;
+    private BiFunction<String, ? super ClientOptionsBuilder, ClientOptionsBuilder> configurator =
+            DEFAULT_CONFIGURATOR;
+
+    /**
+     * Creates a {@link ArmeriaRetrofitBuilder} with the default {@link ClientFactory}.
+     */
+    public ArmeriaRetrofitBuilder() {
+        this(ClientFactory.DEFAULT);
+    }
+
+    /**
+     * Creates a {@link ArmeriaRetrofitBuilder} with the specified {@link ClientFactory}.
+     */
+    public ArmeriaRetrofitBuilder(ClientFactory clientFactory) {
+        requireNonNull(clientFactory, "factory");
+        this.clientFactory = clientFactory;
```

Comment:
nit. Can merge this line to above line.
`this.clientFactory = requireNonNull(clientFactory, "clientFactory");`